admins/admins-router.js

```diff
@@ -1,13 +1,38 @@
-const router = require('express').Router();
+const router = require("express").Router();
 
-const getAllAdmins = require('./getAllAdmins');
-const getAdmin = require('./getAdmin');
-const createCompany = require('../companies/createCompany');
-const validateAdminId = require('../middlewares/validateAdminId');
-const validateCompany = require('../middlewares/validateCompany');
+// GET helpers
+const getAllAdmins = require("./getAllAdmins");
+const getAdmin = require("./getAdmin");
+const getCompanies = require('../admins/getCompanies');
+const getACompany = require('../admins/getACompany')
 
-router.get('/', getAllAdmins);
-router.get('/:id', validateAdminId, getAdmin);
-router.post('/:id/add-company', validateAdminId, validateCompany, createCompany);
+// POST helpers
+const createCompany = require("../companies/createCompany");
+const createJobListing = require("../jobListings/createJobListing");
 
-module.exports = router;
\ No newline at end of file
+// Validation helpers
+const validateAdminId = require("../middlewares/validateAdminId");
+const validateCompany = require("../middlewares/validateCompany");
+const validateJobListing = require("../middlewares/validateJobListing");
+const validateCompanyId = require('../middlewares/validateCompanyId')
+
+// PUT helpers
+const deleteJobListing = require("../jobListings/deleteJobListing");
+const updateJobListing = require("../jobListings/updateJobListing");
+
+
+// Admins
+router.get("/", getAllAdmins);
+router.get("/:id", validateAdminId, getAdmin);
+
+// Admins => Companies
+router.post("/:id/add-company", validateAdminId, validateCompany, createCompany);
+router.get('/:id/companies', validateAdminId, getCompanies);
+router.get('/:id/companies/:companyId', validateCompanyId, validateAdminId, getACompany);
+
+// Admins => Companies => Listings
+router.post("/:id/add-joblisting", validateJobListing, validateAdminId, createJobListing);
+router.delete("/:id/delete-joblisting/:jobListing_id", validateAdminId, deleteJobListing);
+router.put("/:id/update-joblisting/:jobListing_id", validateAdminId, updateJobListing);
```

Comment:
This endpoints looks wrong. If you look at the database we modeled, joblistings have no link directly with an admin, so you can't use `/:id/` from the admins endpoint to add a listing. You have to get the profile of the company first and add the listing to that company profile. I have completed the endpoint to get a company profile `/:id/companies/:companyId`. To add a listing, you have to attach it to this endpoint like so, `/:id/companies/:companyId/addListing`.